packages/hardhat-core/src/internal/hardhat-network/provider/utils/convertToRethnet.ts

```diff
@@ -476,3 +480,64 @@ function isEip2930SignedTransaction(
     !isEip1559SignedTransaction(transaction) && "oddYParity" in transaction
   );
 }
+
+export function rethnetRpcDebugTraceToHardhat(
+  rpcDebugTrace: DebugTraceResult
+): RpcDebugTraceOutput {
+  const structLogs = rpcDebugTrace.structLogs.map((log) => {
+    const result: RpcStructLog = {
+      depth: Number(log.depth),
+      gas: Number(log.gas),
+      gasCost: Number(log.gasCost),
+      op: log.opName!,
+      pc: Number(log.pc),
+    };
+
+    if (log.memory !== undefined) {
+      result.memory = log.memory;
+    }
+
+    if (log.stack !== undefined) {
+      // Remove 0x prefix which is required by EIP-3155, but not expected by Hardhat.
+      result.stack = log.stack?.map((item) => item.slice(2));
+    }
+
+    if (log.storage !== undefined) {
+      result.storage = Object.fromEntries(
+        Object.entries(log.storage).map(([key, value]) => {
+          return [key.slice(2), value.slice(2)];
+        })
+      );
+    }
+
+    if (log.error !== undefined) {
+      result.error = {
+        message: log.error,
+      };
+    }
+
+    return result;
+  });
+
+  // REVM trace adds initial STOP that Hardhat doesn't expect
+  if (structLogs.length > 0 && structLogs[0].op === "STOP") {
+    structLogs.shift();
+  }
+
+  return {
+    failed: !rpcDebugTrace.pass,
```

Comment:
It's just boolean negation. `pass` is true if the transaction succeeded and false otherwise, and we want to negate that to make `failed` true if the transaction didn't succeed. I'm using `pass` in the Rust implementation, because that's what EIP-3155 uses as well.

Reply:
🤦